src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -15,6 +15,8 @@
 use PhpCsFixer\OptionsResolver;
 
 /**
+ * Maybe better the other way around, i.e. now there is no upgrade path to the new interface without deprecation warnings
```

Comment:
Agreed, but if we deprecate this in 3.0, that means that integrating these changes won't be actually complete until 4.0, right? Isn't there another solution so that all deprecations can be removed in 3.0 without adding others?